src/main/java/com/gregtechceu/gtceu/core/mixins/WorldLoaderMixin.java

```diff
@@ -0,0 +1,31 @@
+package com.gregtechceu.gtceu.core.mixins;
+
+import com.gregtechceu.gtceu.GTCEu;
+
+import net.minecraft.core.RegistryAccess;
+import net.minecraft.server.WorldLoader;
+
+import com.llamalad7.mixinextras.sugar.Local;
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+
+@Mixin(value = WorldLoader.class)
+public class WorldLoaderMixin {
+
+    @Inject(method = "load",
+            at = @At(value = "INVOKE",
+                     target = "Lnet/minecraft/server/ReloadableServerResources;loadResources(Lnet/minecraft/server/packs/resources/ResourceManager;Lnet/minecraft/core/RegistryAccess$Frozen;Lnet/minecraft/world/flag/FeatureFlagSet;Lnet/minecraft/commands/Commands$CommandSelection;ILjava/util/concurrent/Executor;Ljava/util/concurrent/Executor;)Ljava/util/concurrent/CompletableFuture;"))
+    private static <D, R> void gtceu$injectLoad(WorldLoader.InitConfig initConfig,
```

Comment:
I think I found a way to do this that isn't COMPLETE Susge
This RegistryAccess object that we're trying to get is the exact same RegistryAccess that gets sent when the `addReloadListenerEvent` event is fired, and we can freely grab that one and store it.
Preliminary testing shows that it works fine without nulling out or crashing, and ore veins/fluid veins are still present. Seems to work server/client too.
Maybe ask screret.

Reply:
it does make sense, I just finished following the origin of the frozen registry, it sends the exact same registry access object that the mixin was saving and that's the only place that triggers this event.